### PR TITLE
CLN: remove block._coerce_values

### PR DESCRIPTION
**Pull request by jbrockmendel**
viable now that we stopped coercing timedelta and datetime to i8

I'd like to clean up DatetimeTZBlock._try_coerce_result before getting this in.  Part of that means pushing the handling down to the function being wrapped

### CODE REVIEW

**Comment by jbrockmendel** [comment]
#27445 is a subset of this

File: pandas/tests/indexing/test_datetime.py [hunk 1]
```diff
@@ -51,15 +51,15 @@ def test_indexing_with_datetime_tz(self):
         # indexing
         result = df.iloc[1]
         expected = Series(
-            [Timestamp("2013-01-02 00:00:00-0500", tz="US/Eastern"), np.nan, np.nan],
```
Comment (jreback) [inline pandas/tests/indexing/test_datetime.py]: does this not work now (the existing code)?

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -2512,22 +2468,22 @@ def _try_coerce_args(self, other):
     def _try_coerce_result(self, result):
         """ reverse of try_coerce_args """
         if isinstance(result, np.ndarray):
-            if result.dtype.kind in ["i", "f"]:
-                result = result.astype("M8[ns]")
+            if result.ndim == 2:
```
Comment (jreback) [inline pandas/core/internals/blocks.py]: can you add a comment or 2 here

File: pandas/core/internals/blocks.py [hunk 3]
```diff
@@ -1547,14 +1530,13 @@ def quantile(self, qs, interpolation="linear", axis=0):
             # We need to operate on i8 values for datetimetz
             # but `Block.get_values()` returns an ndarray of objects
             # right now. We need an API for "values to do numeric-like ops on"
-            values = self.values.asi8
+            values = self.values.view("M8[ns]")
```
Comment (jreback) [inline pandas/core/internals/blocks.py]: we should probably remove .asi8 on Index as I agree its sligthly confusing (I don't think we need deprecation)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_datetime.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/test_datetime.py]: I think the test is incorrect, and tm.assert_series_equal is getting it wrong.
```
now = pd.Timestamp.now("US/Eastern")
left = pd.Series([now, pd.NaT, pd.NaT], dtype=object)
right = pd.Series([now, np.nan, np.nan], dtype=object)

tm.assert_series_equal(left, right)
```

Comment (jbrockmendel) [reply pandas/tests/indexing/test_datetime.py]: Will open an issue for assert_series_equal

File: pandas/core/internals/blocks.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py]: Done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**